dagwriter.go

```diff
@@ -0,0 +1,79 @@
+package dagwriter
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+
+	blocks "github.com/ipfs/go-block-format"
+	blockstore "github.com/ipfs/go-ipfs-blockstore"
+	"github.com/ipld/go-ipld-prime"
+	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
+)
+
+type DagWriter interface {
+	Store(lnkCtx ipld.LinkContext, lp ipld.LinkPrototype, n ipld.Node) (ipld.Link, error)
```

Comment:
will this fully recurse storing links in the node? what if there's a link that can't be resolved? is there a variant with a selector /walker?

Reply:
What does "what if there's a link that can't be resolved?" mean? This is purely storage -- we're not loading/resolving anything here.

Reply:
Lets say the node I provide to store is a map containing a link. Will that link be followed?